src/genn/backends/cuda/backend.cc

```diff
@@ -55,187 +55,170 @@ const EnvironmentLibrary::Library doubleRandomFunctions = {
+//! Copy entire array to device
+void Array::pushToDevice()
+{
+    if(!(getLocation() & VarLocationAttribute::DEVICE) || !(getLocation() & VarLocationAttribute::HOST)) {
+        throw std::runtime_error("Cannot push array that isn't present on host and device");
     }
 
-    //! Copy a 1D slice of elements to device 
-    /*! \param offset   Offset in elements to start copying from
-        \param count    Number of elements to copy*/
-    virtual void pushSlice1DToDevice(size_t offset, size_t count) final
-    {
-        if(!(getLocation() & VarLocationAttribute::DEVICE) || !(getLocation() & VarLocationAttribute::HOST)) {
-            throw std::runtime_error("Cannot push array that isn't present on host and device");
-        }
-
-        if(!(getLocation() & VarLocationAttribute::ZERO_COPY)) {
-            // If end of slice overflows array, give error
-            if((offset + count) > getCount()) {
-                throw std::runtime_error("Cannot pull slice that overflows array");
-            }
+    if(!(getLocation() & VarLocationAttribute::ZERO_COPY)) {
+        CHECK_CUDA_ERRORS(cudaMemcpy(getDevicePointer(), getHostPointer(), getSizeBytes(), cudaMemcpyHostToDevice));
+    }
+}
 
-            // Convert offset and count to bytes and copy
-            const size_t offsetBytes = offset * getType().getValue().size;
-            const size_t countBytes = count * getType().getValue().size;
-            CHECK_CUDA_ERRORS(cudaMemcpy(getDevicePointer() + offsetBytes, getHostPointer() + offsetBytes, 
-                                         countBytes, cudaMemcpyHostToDevice));
-        }
+//! Copy entire array from device
+void Array::pullFromDevice()
+{
+    if(!(getLocation() & VarLocationAttribute::DEVICE) || !(getLocation() & VarLocationAttribute::HOST)) {
+        throw std::runtime_error("Cannot pull array that isn't present on host and device");
     }
 
-    //! Copy a 1D slice of elements from device 
-    /*! \param offset   Offset in elements to start copying from
-        \param count    Number of elements to copy*/
-    virtual void pullSlice1DFromDevice(size_t offset, size_t count) final
-    {
-        if(!(getLocation() & VarLocationAttribute::DEVICE) || !(getLocation() & VarLocationAttribute::HOST)) {
-            throw std::runtime_error("Cannot pull array that isn't present on host and device");
-        }
+    if(!(getLocation() & VarLocationAttribute::ZERO_COPY)) {
+        CHECK_CUDA_ERRORS(cudaMemcpy(getHostPointer(), getDevicePointer(), getSizeBytes(), cudaMemcpyDeviceToHost));
+    }
+}
 
-        if(!(getLocation() & VarLocationAttribute::ZERO_COPY)) {
-            // If end of slice overflows array, give error
-            if((offset + count) > getCount()) {
-                throw std::runtime_error("Cannot pull slice that overflows array");
-            }
+//! Copy a 1D slice of elements to device 
+/*! \param offset   Offset in elements to start copying from
+    \param count    Number of elements to copy*/
```

Comment:
these comments get used to generate API documentation, can you move them to the header?